Source/missiles.cpp

```diff
@@ -3510,21 +3516,6 @@ void ProcessWeaponExplosion(Missile &missile)
 
 	missile.duration--;
 	const Player &player = Players[missile._misource];
-	int mind;
-	int maxd;
-	DamageType damageType;
-	if (missile.var2 == 1) {
-		// BUGFIX: damage of missile should be encoded in missile struct; player can be dead/have left the game before missile arrives.
-		mind = player._pIFMinDam;
-		maxd = player._pIFMaxDam;
-		damageType = DamageType::Fire;
-	} else {
-		// BUGFIX: damage of missile should be encoded in missile struct; player can be dead/have left the game before missile arrives.
-		mind = player._pILMinDam;
-		maxd = player._pILMaxDam;
-		damageType = DamageType::Lightning;
-	}
-	CheckMissileCol(missile, damageType, mind, maxd, false, missile.position.tile, false);

```

Comment:
Would moving this code under the `if (missile.var1 == 0) {` condition also fix the bug?

Reply:
That certainly would prevent multiple collision checks, but the call chain is `DoAttack() > AddMissile() > AddWeaponExplosion()` meaning the collision happens as a direct result of the attack connecting in `DoAttack()`. On the other hand, `ProcessWeaponExplosion()` is called during `ProcessMissiles()` which means they don't even happen during the same step of the game logic, although they do happen during the same game tick thanks to order of events.

TBH, this probably doesn't really matter, but it does stand to reason that the missile should attempt to hit the monster that the melee attack connected with so it makes more sense to me to do it immediately after the melee attack connects. This is also more consistent with how elemental arrows perform both collision checks.

Reply:
Thanks for the explanation 🙂 . 
I was curious about the motivation behind it.